src/components/RichSelectField/index.tsx

```diff
@@ -0,0 +1,144 @@
+import { RichSelect } from '@scaleway/ui'
+import React, { Children, ComponentProps, ReactElement, useMemo } from 'react'
+import { useField } from 'react-final-form'
+import pickValidators from '../../helpers/pickValidators'
+import useValidation from '../../hooks/useValidation'
+import { BaseFieldProps } from '../../types'
+
+type RichSelectProps = ComponentProps<typeof RichSelect>
+type RichSelectOptionProps = ComponentProps<typeof RichSelect.Option>
+type RichSelectOptionElement = ReactElement<RichSelectOptionProps>
+type RichSelectOptions = ComponentProps<typeof RichSelect>['options']
+type RichSelectOptionOrGroup = NonNullable<RichSelectOptions>[number]
+type RichSelectOption = { value: string; label: string }
+
+export type RichSelectFieldProps<
+  T extends RichSelectOptionOrGroup = RichSelectOptionOrGroup,
+> = BaseFieldProps<T> & {
+  className?: string
+  disabled?: boolean
+  id?: string
+  name: string
+  required?: boolean
+  requiredMessage?: string
+  children?: RichSelectOptionElement | RichSelectOptionElement[]
+  placeholder?: string
+  error?: boolean | string
+  onChange?: (newValue: RichSelectOption | RichSelectOption[]) => void
+  onBlur?: RichSelectProps['onBlur']
+  onFocus?: RichSelectProps['onFocus']
+  options?: RichSelectOptions
+}
+
+const identity = <T,>(x: T) => x
+
+const RichSelectField = <
+  T extends RichSelectOptionOrGroup = RichSelectOptionOrGroup,
+>({
+  children,
+  className,
+  error,
+  format = identity as NonNullable<BaseFieldProps<T>['format']>,
+  id,
+  multiple,
+  name,
+  onBlur,
+  onChange,
+  onFocus,
+  options: optionsProp,
+  parse = identity as NonNullable<BaseFieldProps<T>['parse']>,
+  placeholder,
+  required,
+  value,
+}: RichSelectFieldProps<T>) => {
+  const validate = useValidation({
+    validators: pickValidators<T>({
+      minLength: multiple ? 1 : undefined,
```

Comment:
minLength can be undefined if multiple but not required 🤔

Reply:
You can have maxLength too

Reply:
fixed